### ISSUE TITLE
Extension slowing down the website

### ISSUE THREAD

**Issue by s0upmandev** [0x48piraj/fadblock#56]
I've noticed that ever since installing this extension, it has slowed down Youtube significantly, even crashing on multiple occasions

**Comment by 0x48piraj** [comment]
Yes, there have been some compatibility and performance issues lately with the new API.

As of now I am applying a quick monkey-patch to fix the issue while investigating the cause for this.

**Comment by ericbig66** [comment]
I think there are also problems in chrome,  it takes about 5~10 seconds just to load the page, and start to play video at about 15 second, also shrot experience are ruined, due to the lag of loading, it would be lag for about 2~3 second after scroll to next video, maybe try to just add a feature to disable ad block on Shorts, though it would have ad between shorts, but smoother short experience is more important, currently my solution is constantly delete cookies generated by YT to speed up the loading process.. By the way my Internet speed is not slow at all, ping is about 6 ms, with 275Mbps DL and 93Mbps UL

**Comment by 0x48piraj** [comment]
FadBlock doesn't use any cookies but that's a good idea, disabling ad blocks on shorts. I'll look into that, but first, 2.0 is going to release on Chrome and I want to know if the new API reduces any load time or not before working on performance optimizations.

**Comment by 0x48piraj** [comment]
As all of you might be aware, YouTube is injecting a lot of DOM and custom scripts to detect adblockers which in turn are slowing their own website. [[1](https://www.reddit.com/r/youtube/comments/17z8hsz/youtube_has_started_to_artificially_slow_down/)], [[2](https://www.reddit.com/r/youtube/comments/16fa8cu/is_youtube_loading_slow_for_anyone_else/)] and many more.

I am working on some optimizations but it will need heavy testing as it can be a breaking change and the userbase is very volatile right now so I have delayed the patch for two updates. As reported in other threads, once YouTube is loaded, the lag should go away significantly, is that the case with you as well?

**Comment by 0x48piraj** [comment]
As all of you might be aware, YouTube is injecting a lot of DOM and custom scripts to detect adblockers which in turn are slowing their own website. [[1](https://www.reddit.com/r/youtube/comments/17z8hsz/youtube_has_started_to_artificially_slow_down/)], [[2](https://www.reddit.com/r/youtube/comments/16fa8cu/is_youtube_loading_slow_for_anyone_else/)] and many more.

I am working on some optimizations but it will need some testing as it can be a breaking change and the userbase is very volatile right now (thankfully the v2.6 is working smoothly as it seems) so I have delayed the patch for two updates.

As reported in other threads, once YouTube is loaded, the lag should go away significantly. All in all, it's mostly YouTube's own doing.
